sources/packages/backend/apps/queue-consumers/src/services/cas-invoice/cas-invoice.service.ts

```diff
@@ -0,0 +1,264 @@
+          },
+          casInvoiceDetails: {
+            id: true,
+            valueAmount: true,
+            casDistributionAccount: {
+              id: true,
+              operationCode: true,
+              awardValueCode: true,
+              distributionAccount: true,
+            },
+          },
+        },
+        relations: {
+          casInvoiceBatch: true,
+          disbursementReceipt: true,
+          casSupplier: true,
+          casInvoiceDetails: {
+            casDistributionAccount: true,
+          },
+        },
+        where: {
+          casInvoiceBatch: {
+            approvalStatus: CASInvoiceBatchApprovalStatus.Approved,
+          },
+          invoiceStatus: CASInvoiceStatus.Pending,
+        },
+      });
+    return pendingInvoices;
+  }
+
+  /**
+   * Send pending invoices to CAS.
+   * @param parentProcessSummary parent process summary for logging.
+   * @param pendingInvoices pending invoices.
+   */
+
+  async sendInvoices(
+    parentProcessSummary: ProcessSummary,
+    pendingInvoices: CASInvoice[],
+  ): Promise<void> {
+    // Process each invoice in parallel.
+    await processInParallel(
+      (pendingInvoice) =>
+        this.processInvoice(pendingInvoice, parentProcessSummary),
+      pendingInvoices,
+    );
+  }
+
+  /**
+   * Process and send an invoice to CAS.
+   * @param pendingInvoice pending invoice.
+   * @param parentProcessSummary parent process summary for logging.
+   */
+  private async processInvoice(
+    pendingInvoice: CASInvoice,
+    parentProcessSummary: ProcessSummary,
+  ): Promise<void> {
+    // Log information about the pending invoice being processed.
+    parentProcessSummary.info(
+      `Processing pending invoice: ${pendingInvoice.invoiceNumber}.`,
+    );
+    const pendingInvoicePayload = this.getPendingInvoicePayload(pendingInvoice);
+    parentProcessSummary.info(
+      `Pending invoice payload invoice number: ${pendingInvoicePayload.invoiceNumber}.`,
+    );
+    let response: SendPendingInvoicesResponse;
+    try {
+      response = await this.casService.sendPendingInvoices(
+        pendingInvoicePayload,
+      );
+      response.invoiceNumber
+        ? parentProcessSummary.info(
+            `Invoice sent to CAS ${response.casReturnedMessages}.`,
+          )
+        : parentProcessSummary.warn(
+            `Invoice ${pendingInvoicePayload.invoiceNumber} send to CAS did not succeed. Reason: ${response.casReturnedMessages}.`,
+          );
+      await this.dataSource.getRepository(CASInvoice).update(
+        {
+          id: pendingInvoice.id,
+        },
+        {
+          invoiceStatus: CASInvoiceStatus.Sent,
+          invoiceStatusUpdatedOn: new Date(),
+          modifier: { id: this.systemUsersService.systemUser.id },
+        },
+      );
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
+        if (error.name === CAS_BAD_REQUEST) {
+          parentProcessSummary.warn("Known CAS error while sending invoice.");
```

Comment:
Is this log not equivalent to the one already present in `processBadRequestErrors`?